lib/src/main/kotlin/com/swmansion/starknet/data/types/StarknetByteArray.kt

```diff
@@ -0,0 +1,27 @@
+package com.swmansion.starknet.data.types
+
+import com.swmansion.starknet.data.types.conversions.ConvertibleToCalldata
+import com.swmansion.starknet.extensions.splitToShortStrings
+import com.swmansion.starknet.extensions.toFelt
+
+data class StarknetByteArray(
+    val data: List<Felt>,
+    val pendingWord: Felt,
+    val pendingWordLen: Int,
+) : ConvertibleToCalldata {
+    override fun toCalldata(): List<Felt> {
+        return listOf(data.size.toFelt) + data + listOf(pendingWord, pendingWordLen.toFelt)
```

Comment:
can you give a bit more context in here? what is a pending word? and shouldn't the length be the first element (before pendingWord)? (maybe I just misunderstand/do not know something, hence my question)

Reply:
You can check out explanation of `ByteArray` fields and its serialization process in the [docs](https://docs.starknet.io/documentation/architecture_and_concepts/Smart_Contracts/serialization_of_Cairo_types/#serialization_of_byte_arrays).

Should we add a docstring that mentions that this class corresponds to `ByteArray` struct in Cairo, so there's no confusion about its purpose?

Reply:
yes, definitely let's add this